streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -92,33 +97,36 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
         }
 
         @Override
-        public void process(final K key, final V value) {
+        public void process(final Record<KIn, VIn> record) {
             // if the key is null, then ignore the record
-            if (key == null) {
-                LOG.warn(
+            if (record.key() == null) {
+                context.recordMetadata().ifPresent(recordMetadata -> LOG.warn(
                     "Skipping record due to null key. topic=[{}] partition=[{}] offset=[{}]",
-                    context().topic(), context().partition(), context().offset()
-                );
+                    recordMetadata.topic(), recordMetadata.partition(), recordMetadata.offset()
+                ));
                 droppedRecordsSensor.record();
                 return;
             }
 
             if (queryableName != null) {
-                final ValueAndTimestamp<V> oldValueAndTimestamp = store.get(key);
-                final V oldValue;
+                final ValueAndTimestamp<VIn> oldValueAndTimestamp = store.get(record.key());
+                final VIn oldValue;
                 if (oldValueAndTimestamp != null) {
                     oldValue = oldValueAndTimestamp.value();
-                    if (context().timestamp() < oldValueAndTimestamp.timestamp()) {
-                        LOG.warn("Detected out-of-order KTable update for {} at offset {}, partition {}.",
-                            store.name(), context().offset(), context().partition());
+                    if (record.timestamp() < oldValueAndTimestamp.timestamp()) {
+                        context.recordMetadata().ifPresent(recordMetadata ->
+                            LOG.warn(
+                                "Detected out-of-order KTable update for {} at offset {}, partition {}.",
```

Comment:
Similar here. If it's absent, maybe we can say `"Detected out-of-order KTable update for {}. Partition and offset not known."`

Reply:
Also, this is not related to your change, but it occurs to me that we're missing some important information here, which would likely save people a lot of time. Can we add the new and old timestamps to the log messages?